src/os/inc/osapi-os-core.h

```diff
@@ -1505,4 +1505,59 @@ void OS_printf_disable(void);
 void OS_printf_enable(void);
 /**@}*/
 
+
```

Comment:
Curious on thoughts about these in osapi-os-core.h.  I'd think they'd be candidates for a separate or different header.

Reply:
Yes, these could be split into a separate header file.

Note that OSAL is "caching" the value from the BSP here, so the API really is still an OSAL API, not a BSP API. The BSP sets the cache when it boots and the API just reads it and returns it to the app. I was hoping to avoid having an actual BSP API - the goal should be to hide the BSP details as much as we can, not expose it.

FWIW, we should also consider making the `inc` directory structure more like other modules where we have the public API in `fsw/inc` and the source in `fsw/src`, rather than `src/os/inc` where OSAL has it right now. That would be a follow-on change but this may actually make it easier by removing at least some build system references to `osal/src/os/inc`.

Reply:
Thanks, that background helps... historically I've broken includes into smaller "topics" to limit conceptual impact of changes, but I realize OSAL has an established pattern.  It has paid off for me in relation to review panels and operational software support processes.

I think cFS 7.0 is due a major directory restructure... very interested in an in-depth design/review CCB discussion on the topic.